host/class/hid/usb_host_hid/hid_host.c

```diff
@@ -424,50 +461,42 @@ static void hid_host_notify_interface_connected(hid_device_t *hid_device)
 static esp_err_t hid_host_interface_list_create(hid_device_t *hid_device,
         const usb_config_desc_t *config_desc)
 {
+    assert(hid_device);
+    assert(config_desc);
     size_t total_length = config_desc->wTotalLength;
     const usb_intf_desc_t *iface_desc = NULL;
     const hid_descriptor_t *hid_desc = NULL;
-    const usb_ep_desc_t *ep_desc = NULL;
     const usb_ep_desc_t *ep_in_desc = NULL;
-    int offset = 0;
+    int iface_offset = 0;
+    int hid_desc_offset = 0;
 
+    // Get first Interface descriptor
+    iface_desc = GET_NEXT_INTERFACE_DESC(config_desc, total_length, iface_offset);
     // For every Interface
-    for (int i = 0; i < config_desc->bNumInterfaces; i++) {
-        iface_desc = usb_parse_interface_descriptor(config_desc, i, 0, &offset);
+    while (iface_desc != NULL) {
+
         hid_desc = NULL;
+        hid_desc_offset = iface_offset;
         ep_in_desc = NULL;
 
         if (USB_CLASS_HID == iface_desc->bInterfaceClass) {
-            // HID descriptor
-            hid_desc = (const hid_descriptor_t *) usb_parse_next_descriptor_of_type((const usb_standard_desc_t *) iface_desc,
-                       total_length,
-                       HID_CLASS_DESCRIPTOR_TYPE_HID,
-                       &offset);
-            if (!hid_desc) {
-                return ESP_ERR_NOT_FOUND;
-            }
-
-            // EP descriptors for Interface
-            for (int i = 0; i < iface_desc->bNumEndpoints; i++) {
-                int ep_offset = 0;
-                ep_desc = usb_parse_endpoint_descriptor_by_index(iface_desc, i, total_length, &ep_offset);
-
-                if (ep_desc) {
-                    if (USB_EP_DESC_GET_EP_DIR(ep_desc)) {
-                        ep_in_desc = ep_desc;
-                    }
-                } else {
-                    return ESP_ERR_NOT_FOUND;
+            ESP_LOGD(TAG, "Found HID, bInterfaceNumber=%d, offset=%d, total_len=%d",
```

Comment:
is `total_len` relevant here?

Reply:
No, but it is for debug purpose, to be sure that `offset` is not more that `total_len`
Probably, I can remove this line...

Reply:
OK, thanks! your call :)